runtime/vm/ContinuationHelpers.cpp

```diff
@@ -317,6 +323,8 @@ freeContinuation(J9VMThread *currentThread, j9object_t continuationObject, BOOLE
 void
 recycleContinuation(J9JavaVM *vm, J9VMThread *vmThread, J9VMContinuation* continuation, BOOLEAN skipLocalCache)
```

Comment:
do we need `skipLocalCache` if the vmthread is NULL? Is there a case where we would want to skip local cache, but have a valid vmthread in hand?

Reply:
If `freeContinuation` is called by GC, then it will pass down the currentThread into `recycleContinuation`, in such case we cant tell if the currentThread is a carrier thread or not with out expensive checks on the java thread object